tests/src/Kernel/Validators/ValidatorEmptyCellProcessTest.php

```diff
@@ -0,0 +1,452 @@
+   */
+  protected static $modules = [
+    'system',
+    'user',
+    'path',
+    'path_alias',
+    'views',
+    'field',
+    'field_ui',
+    'markup',
+    'field_group',
+    'tripal',
+    'tripal_chado',
+    'tripal_layout',
+    'trpcultivate',
+  ];
+
+  /**
+   * Drupal render service.
+   *
+   * @var Drupal\Core\Render\RendererInterface
+   */
+  protected $renderer;
+
+  /**
+   * An instance of the validator.
+   *
+   * @var Drupal\trpcultivate\Plugin\Validators\EmptyCell
+   */
+  protected $validator_instance;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    // Ensure we see all logging in tests.
+    \Drupal::state()->set('is_a_test_environment', TRUE);
+
+    $validator_id = 'empty_cell';
+    $this->validator_instance = $this->container
+      ->get('plugin.manager.trpcultivate_validator')
+      ->createInstance($validator_id);
+
+    $this->renderer = $this->container->get('renderer');
+  }
+
+  /**
+   * Data Provider for testProcessEmptyCellFailures().
+   *
+   * @return array
+   *   Each scenario is an array with the following:
+   *   - The failures array that gets passed to the process method. It contains
+   *     the following keys:
+   *     - The line number that triggered this failed validation status.
+   *       - 'case': a developer-focused string describing the case checked.
+   *       - 'valid': FALSE to indicate that validation failed.
+   *       - 'failedItems': array of items that failed with the following keys:
+   *         - 'empty_indices': A list of column indices in the line which were
+   *           checked and found to be empty.
+   *   - An array of tokens to use for altering the messages that get displayed
+   *     to the user. The key is the token, (ex. 'project'), and the value is
+   *     the new value to be shown for that token.
+   *   - An array of additional metadata (or contextual information) needed by
+   *     the process method. Here, the following keys are expected:
+   *     - 'column_headers': This contains an array of headers. The index in
+   *       this array MUST match the position (starting with 0) of the column in
+   *       the input file.
+   *       Eg: 'column_headers' => [
+   *             '2' => 'Header 1', // Header of column #3
+   *             '4' => 'Header 2', // Header of column #5
+   *           ].
+   *   - An array of expectations that we want to find in the resulting rendered
+   *     output which has the following keys:
+   *     - 'expected_message': The message expected in the return value of the
+   *       process method for this scenario.
+   *     - 1+ arrays keyed by the line number in the input file that triggered
+   *       the failed validation status, further keyed by:
+   *       - 'expected_columns': A comma-separated list of column headers that
+   *         map to the expected columns with empty values.
+   */
+  public static function provideEmptyCellFailedCases() {
```

Comment:
You should add a test case here to make sure tokens are working. In this case, a test for substituting the message for 'case-empty-value'.